Localization/cubature_kalman_filter/cubature_kalman_filter.py

```diff
@@ -0,0 +1,204 @@
+from scipy.linalg import sqrtm
+
+
+dt = 0.1
+N = 100
+
+show_final = 1
+show_animation = 0
+show_ellipse = 0
+
+
+z_noise = np.array([[0.1, 0.0, 0.0, 0.0],
+                    [0.0, 0.1, 0.0, 0.0],
+                    [0.0, 0.0, 0.1, 0.0],
+                    [0.0, 0.0, 0.0, 0.1]])
+
+
+x_0 = np.array([[0.0],                                  # x position    [m]
+                [0.0],                                  # y position    [m]
+                [0.0],                                  # yaw           [rad]
+                [1.0],                                  # velocity      [m/s]
+                [0.1]])                                 # yaw rate      [rad/s]
+
+
+p_0 = np.array([[1e-3, 0.0, 0.0, 0.0, 0.0],
+                [0.0, 1e-3, 0.0, 0.0, 0.0],
+                [0.0, 0.0, 1.0, 0.0, 0.0],
+                [0.0, 0.0, 0.0, 1.0, 0.0],
+                [0.0, 0.0, 0.0, 0.0, 1.0]])
+
+
+q = np.array([[1e-11, 0.0,    0.0,               0.0, 0.0],
+              [0.0, 1e-11,    0.0,               0.0, 0.0],
+              [0.0, 0.0,    np.deg2rad(1e-4),   0.0, 0.0],
+              [0.0, 0.0,    0.0,               1e-4, 0.0],
+              [0.0, 0.0,    0.0,                0.0, np.deg2rad(1e-4)]])
+
+
+hx = np.array([[1.0, 0.0, 0.0, 0.0, 0.0],
+               [0.0, 1.0, 0.0, 0.0, 0.0],
+               [0.0, 0.0, 0.0, 1.0, 0.0],
+               [0.0, 0.0, 0.0, 0.0, 1.0]])
+
+
+r = np.array([[0.015, 0.0, 0.0, 0.0],
+              [0.0, 0.010, 0.0, 0.0],
+              [0.0, 0.0, 0.1, 0.0],
+              [0.0, 0.0, 0.0, 0.01]])**2
+
+
+def main():
+    print(__file__ + " start!!")
+    x_est = x_0
+    p_est = p_0
+    x_true = x_0
+    x_true_cat = np.array([x_0[0, 0], x_0[1, 0]])
+    x_est_cat = np.array([x_0[0, 0], x_0[1, 0]])
+    z_cat = np.array([x_0[0, 0], x_0[1, 0]])
+    for i in range(N):
+        x_true = f(x_true)
+        z = gen_measurement(x_true)
+        if i == (N - 1) and show_final == 1:
+            show_final_flag = 1
+        else:
+            show_final_flag = 0
+        if show_animation == 1:
+            plot_animation(i, x_true_cat, x_est_cat, z)
+        if show_ellipse == 1:
+            plot_ellipse(x_est[0:2], p_est)
+        if show_final_flag == 1:
+            plot_final(x_true_cat, x_est_cat, z_cat)
+        x_est, p_est = cubature_kalman_filter(x_est, p_est, z)
+        x_true_cat = np.vstack((x_true_cat, x_true[0:2].T))
+        x_est_cat = np.vstack((x_est_cat, x_est[0:2].T))
+        z_cat = np.vstack((z_cat, z[0:2].T))
+    print('CKF Over')
+
+
+def cubature_kalman_filter(x_est, p_est, z):
+    x_pred, p_pred = cubature_prediction(x_est, p_est)
+    x_upd, p_upd = cubature_update(x_pred, p_pred, z)
+    return x_upd, p_upd
+
+
+def f(x):
+    x[0] = x[0] + (x[3]/x[4]) * (np.sin(x[4] * dt + x[2]) - np.sin(x[2]))
+    x[1] = x[1] + (x[3]/x[4]) * (- np.cos(x[4] * dt + x[2]) + np.cos(x[2]))
+    x[2] = x[2] + x[4] * dt
+    x[3] = x[3]
+    x[4] = x[4]
+    return x
+
+
+def h(x):
```

Comment:
Could you please add docstring to explain that this function is for observation model